### PR TITLE
Auth page

### PR DESCRIPTION
**Pull request by bajajcodes**
## Auth Page
- Login and Signup are done using firebase-authentication.
- Guest login feature is also available.
- Toast messages on succesful login and logout.
- Simple alert message on failed login and signup.

## Features
- State managment is done using reduxjs/toolkit.
- CheckAuth utility is used for checking login status and redirecting 
  for RequireAuth utility (upcoming).
- Hooks and Utilities are used for code reusability and modularity.
- Hook useValidateAuthForm is used for dispatching succesful signup/login 
  and informing about failed attempt.
- FormUtil is used for pure reusability purpose.

## Redux Toolkit
- Auth Slice is used for auth state managment, doing signup/login 
  api calls using createAsyncThunk and storiing/getting/clearing
  token from local storage using Local Storage utility.
  
## Deployment Preview
https://deploy-preview-2--3on3doer.netlify.app/

### CODE REVIEW

File: src/features/auth/authSlice.js [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+  async (userInfo, { rejectWithValue }) => {
+    try {
+      const { email, password } = userInfo;
+      const userCredential = await signInWithEmailAndPassword(
+        firebaseLocalAuth,
+        email,
+        password
+      );
+      const user = userCredential.user;
+      return user.uid;
```
Comment (MaruthiPothuganti) [inline src/features/auth/authSlice.js:72]: Succesful login toast misssing as it is mentioned in the PR description

**Review by yaswanthmyneni — APPROVED** [review]
The code is good, you can merge it.

**Review by srishtigautam21 — APPROVED** [review]
The code is clean and understandable
No changes from my side
Good to merge

**Review by arpit01923 — APPROVED** [review]
- clean and impressive code
- good folder structure and naming convention
- good to merge

**Review by MaruthiPothuganti — COMMENTED** [review]
Clean and well written code. PR looks good to merge

File: src/features/auth/authSlice.js (hunk 1 above)
Comment (bajajcodes) [reply src/features/auth/authSlice.js:72]: As page get redirected, message got hidden.

**Pull request merged**